### ISSUE TITLE
Building v1.1.0 results in a filename collision

### ISSUE THREAD

**Issue by aperture-sandi** [paritytech/polkadot-sdk#1548]
Building v1.1.0 via `RUSTFLAGS="-C target-cpu=native" ~/.cargo/bin/cargo build --release` 
results in a filename collision (below) which results in "this error" when running: `polkadot --chain polkadot --validator`

"this error" -> ```Error:
   0: ESC[91mVersion of worker binary (1.0.0-f60318f6868) is different from node version (1.1.0-f60318f6868), worker_path: ~/polkadot-sdk/target/release/polkadot-prepare-worker. If you ran with `cargo run`, please run `cargo build` first, otherwise try to `cargo clean`. TESTING ONLY: this check can be disabled with --disable-worker-version-checkESC[0m``` 

filename collision...
```
warning: output filename collision.
The bin target `polkadot-execute-worker` in package `polkadot-test-malus v1.0.0 (~/polkadot-sdk/polkadot/node/malus)` has the same output filename as the bin target `polkadot-execute-worker` in package `polkadot v1.1.0 (~/polkadot-sdk/polkadot)`.
Colliding filename is: ~/polkadot-sdk/target/release/polkadot-execute-worker
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
warning: output filename collision.
The bin target `polkadot-execute-worker` in package `polkadot-test-malus v1.0.0 (~/polkadot-sdk/polkadot/node/malus)` has the same output filename as the bin target `polkadot-execute-worker` in package `polkadot v1.1.0 (~/polkadot-sdk/polkadot)`.
Colliding filename is: ~/polkadot-sdk/target/release/polkadot-execute-worker.dwp
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
warning: output filename collision.
The bin target `polkadot-prepare-worker` in package `polkadot-test-malus v1.0.0 (~/polkadot-sdk/polkadot/node/malus)` has the same output filename as the bin target `polkadot-prepare-worker` in package `polkadot v1.1.0 (~/polkadot-sdk/polkadot)`.
Colliding filename is: ~/polkadot-sdk/target/release/polkadot-prepare-worker
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
warning: output filename collision.
The bin target `polkadot-prepare-worker` in package `polkadot-test-malus v1.0.0 (~/polkadot-sdk/polkadot/node/malus)` has the same output filename as the bin target `polkadot-prepare-worker` in package `polkadot v1.1.0 (~/polkadot-sdk/polkadot)`.
Colliding filename is: ~/polkadot-sdk/target/release/polkadot-prepare-worker.dwp
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
```

**Comment by gregorst3** [comment]
I'm having the same problem

**Comment by mrcnski** [comment]
Thanks for the report! It's just a warning and can be ignored. 

More info here: https://github.com/paritytech/polkadot-sdk/issues/1154#issuecomment-1693501588

Closing in favor of #1154. Please feel free to respond there.

**Comment by mrcnski** [comment]
As I noted on #1154, I'll try to fix this by the next release. For now, please just ignore the warning. Apologies!

**Comment by mrcnski** [comment]
Oh, I didn't see the version mismatch error. Yes, that's a hard error. Did you read the release notes? 😉 You'll need to re-visit our installation instructions as they have changed.

**Comment by aperture-sandi** [comment]
Thanks for the reply. The only difference I see here [https://wiki.polkadot.network/docs/maintain-guides-how-to-validate-polkadot](https://wiki.polkadot.network/docs/maintain-guides-how-to-validate-polkadot) is building with the production flag instead of release flag.  Is that our issue?

**Comment by mrcnski** [comment]
Which section there are you following? I tried to update every section but maybe I missed one. The issue is there are now some extra binaries that you have to build as well. The easiest way to do it is `cargo install` in place of `cargo build`. I'd uninstall polkadot first, to avoid any conflicts with cargo's installation.

And yeah, I'd recommend building with the production profile, unless you have a good reason not to. :)

**Comment by aperture-sandi** [comment]
In the #optional-building-the-polkadot-binaries-from-sources section it still says to 
`git clone https://github.com/paritytech/polkadot.git` instead of `polkadot-sdk.git`
When running `cargo install --force --path . --profile production` from /home/<user>/polkadot-sdk I get error:  
```found a virtual manifest at `/home/<user>/polkadot-sdk/Cargo.toml` instead of a package manifest```

**Comment by mrcnski** [comment]
Good catch! ⭐️ The page was updated after new binaries were introduced, but then not after the monorepo migration. I'm on it. For now, You'll have to run `cargo install <...>` from `/home/<user>/polkadot-sdk/polkadot`. Let me know if anything else comes up!

**Comment by bernardoaraujor** [comment]
I'm having a similar issue, trying to build `v1.1.0` for MacOS but `polkadot-execute-worker` binary is still on `v1.0.0`, which prevents the relay from starting

**Comment by mrcnski** [comment]
@bernardoaraujor Some more info would help. How are you building/installing? If the worker binary is out of date, then failing to start a validator node is expected behavior.

**Comment by mrcnski** [comment]
Did you try `cargo clean` as indicated in the error message? I think there is a cargo bug where it sometimes will not re-build the auxiliary binaries, but wiping everything and starting over usually works.

**Comment by mrcnski** [comment]
@bernardoaraujor Did you really start with a fresh git clone? Both workers pull in the same `polkadot_cli::NODE_VERSION` for the version, so it would be very strange if a fresh build resulted in mismatched versions. 🤔 

- https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/src/bin/prepare-worker.rs#L22
- https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/src/bin/execute-worker.rs#L22

**Comment by bernardoaraujor** [comment]
yes, just reproduced again with a fresh git clone

**Comment by wirednkod** [comment]
FWIW I did the same from a fresh clone and got different results:
```
➜ ./target/release/polkadot-execute-worker --version
1.0.0-f60318f6868

➜ ./target/release/polkadot-prepare-worker --version
1.1.0-f60318f6868

➜ ./target/release/polkadot --version
polkadot 1.1.0-f60318f6868

**Comment by mrcnski** [comment]
On the v1.1.0 tag we are still getting the version from an env var instead of a constant:

https://github.com/paritytech/polkadot-sdk/blob/polkadot-v1.1.0/polkadot/src/bin/prepare-worker.rs#L22

So I see two issues here:

1. There is some bug with `SUBSTRATE_CLI_IMPL_VERSION`. It sometimes gets the wrong version, plus it behaves non-deterministically. (cc @bkchr)
2. The PR https://github.com/paritytech/polkadot-sdk/pull/1495 did not make it into v1.1.0. (cc @coderobe)

**Comment by bkchr** [comment]
> There is some bug with SUBSTRATE_CLI_IMPL_VERSION. It sometimes gets the wrong version, plus it behaves non-deterministically. (cc @bkchr)

What does it mean that it gets the wrong version? What version does it return?

**Comment by mrcnski** [comment]
Has someone tried v1.2.0 and is the issue fixed there?

**Comment by mrcnski** [comment]
Going to close as it should be fixed on v1.2.0. Please re-open if not!
